File: C/07.7LED.c
#include <wiringPi.h>
#include <stdio.h>

#define LED1 0                       // wPi. 0(GPIO. 0), BCM. 17, Physical-Pin. 11
#define LED2 2                       // wPi. 2(GPIO. 2), BCM. 27, Physical-Pin. 13
#define LED3 3                       // wPi. 3(GPIO. 3), BCM. 22, Physical-Pin. 15
#define LED4 22                      // wPi. 22(GPIO. 22), BCM. 6, Physical-Pin. 31
#define LED5 23                      // wPi. 23(GPIO. 23), BCM. 13, Physical-Pin. 33
#define LED6 24                      // wPi. 24(GPIO. 24), BCM. 19, Physical-Pin. 35
#define LED7 25                      // wPi. 25(GPIO. 25), BCM. 26, Physical-Pin. 37

int main (void) {

    if (wiringPiSetup() == -1) return -1;

    unsigned char LED[7] = {LED1, LED2, LED3, LED4, LED5, LED6, LED7};  // 배열 안에 담아둠
    unsigned char i;
    unsigned char time = 100;

    for (i = 0 ; i < sizeof(LED) ; i++) {       // LED의 배열 크기만큼 반복
        pinMode(LED[i], OUTPUT);                // OUTPUT 설정
    }

    while (1) {                                 // 무한루프
        for (i = 0; i < sizeof(LED) ; i++) {    // LED의 배열 크기만큼 반복
            digitalWrite(LED[i], HIGH);         // LED[i] 켜짐
            delay(time);                        // time ms 대기
            digitalWrite(LED[i], LOW);          // LED[i] 꺼짐
            time--;                             // time = time - 1

            if (time <= 0) time = 100;          // time이 0이 되면 다시 time을 100으로 돌림
        }
    }
}

File: Python/07.7LED.py
import RPi.GPIO as GPIO             # GPIO 제어 모듈
import time                         # time 모듈

LED1 = 17                           # wPi. 0(GPIO. 0), BCM. 17, Physical-Pin. 11
LED2 = 27                           # wPi. 0(GPIO. 0), BCM. 17, Physical-Pin. 11
LED3 = 22                           # wPi. 0(GPIO. 0), BCM. 17, Physical-Pin. 11
LED4 = 6                            # wPi. 0(GPIO. 0), BCM. 17, Physical-Pin. 11
LED5 = 13                           # wPi. 0(GPIO. 0), BCM. 17, Physical-Pin. 11
LED6 = 19                           # wPi. 0(GPIO. 0), BCM. 17, Physical-Pin. 11
LED7 = 26                           # wPi. 0(GPIO. 0), BCM. 17, Physical-Pin. 11

LED = [LED1, LED2, LED3, LED4, LED5, LED6, LED7]
delay = 0.1                         # 100ms(0.1초)

GPIO.setmode(GPIO.BCM)              # BCM모드 설정

for i in LED :                      # LED의 크기만큼 반복
    GPIO.setup(i, GPIO.OUT)         # OUTPUT 설정
    
try :                               # 정상 작동일 때
    while True :                    # 무한루프
        for i in LED :              # LED의 크기만큼 반복
            GPIO.output(i, True)    # LED[i] 켜짐
            time.sleep(delay)       # delay초 대기
            GPIO.output(i, False)   # LED[i] 꺼짐
            
            delay -= 0.001          # delay = delay - 0.001

            if delay <= 0 :         # delay가 0이 되면(혹은 0보다 작으면)
                delay = 0.1         # 다시 원래 크기대로 복구
            
except :                            # 종료 시
    GPIO.cleanup()                  # GPIO 초기화
    print("end")